src/main/java/frc/robot/commands/DriveWithJoysticks.java

```diff
@@ -85,43 +90,79 @@ public void execute() {
+          : processJoystickAxis(oiRightDriveY.getAsDouble() /* Robot.oi.getSingleDriveAxisRight() */);
+      joystickRight = joysticksReversed ? (baseDrive - processJoystickAxis(oiLeftDriveX.getAsDouble()))
+          : baseDrive + processJoystickAxis(oiLeftDriveX.getAsDouble() /* Robot.oi.getLeftHorizDriveAxis() */);
+      joystickLeft = joysticksReversed ? (baseDrive + processJoystickAxis(oiLeftDriveX.getAsDouble()))
+          : baseDrive - processJoystickAxis(oiLeftDriveX.getAsDouble() /* Robot.oi.getLeftHorizDriveAxis() */);
+
       joystickRight = joystickRight > 1 ? 1 : joystickRight;
-      joystickLeft = baseDrive - processJoystickAxis(oiLeftDriveX.getAsDouble() /* Robot.oi.getLeftHorizDriveAxis() */);
       joystickLeft = joystickLeft > 1 ? 1 : joystickLeft;
       break;
     case Trigger:
-      baseDrive = (oiLeftDriveTrigger.getAsDouble()
-          /* Robot.oi.getLeftTrigger() */ - oiRightDriveTrigger.getAsDouble() /* Robot.oi.getRightTrigger() */ ) * -1;
+      baseDrive = joysticksReversed ? ((oiRightDriveTrigger.getAsDouble() - oiLeftDriveTrigger.getAsDouble()) * -1)
+          : (oiLeftDriveTrigger.getAsDouble()
+              /* Robot.oi.getLeftTrigger() */ - oiRightDriveTrigger.getAsDouble() /* Robot.oi.getRightTrigger() */ )
+              * -1;
       totalTurn = oiLeftDriveX.getAsDouble() /* Robot.oi.getLeftHorizDriveAxis() */
           + (oiRightDriveX.getAsDouble() /* Robot.oi.getRightHorizDriveAxis() */ * rightStickScale);
-      joystickRight = baseDrive + processJoystickAxis(totalTurn);
+
+      joystickRight = joysticksReversed ? (baseDrive - processJoystickAxis(totalTurn))
+          : baseDrive + processJoystickAxis(totalTurn);
+      joystickLeft = joysticksReversed ? (baseDrive + processJoystickAxis(totalTurn))
+          : baseDrive - processJoystickAxis(totalTurn);
+
       joystickRight = joystickRight > 1 ? 1 : joystickRight;
-      joystickLeft = baseDrive - processJoystickAxis(totalTurn);
       joystickLeft = joystickLeft > 1 ? 1 : joystickLeft;
       break;
-    case Sniper:
+    }
 
-      break;
+    if (oiSniperMode.getAsBoolean()) {
+      getMultiplierForSniperMode();
+      double multiplierLevel = getMultiplierForSniperMode();
+      joystickLeft *= multiplierLevel;
+      joystickRight *= multiplierLevel;
     }
+
     driveSubsystem.drive(joystickLeft, joystickRight, alwaysUseHighMaxVel);
+  }
+
+  public double getMultiplierForSniperMode() {
+    double sniperLevel;
+    if (oiHasDualSniperMode) {
+      if (oiSniperHigh.getAsBoolean()) {
+        sniperLevel = oiSniperHighLevel.getAsDouble();
+      } else {
+        sniperLevel = oiSniperLowLevel.getAsDouble();
+      }
+    } else {
+      sniperLevel = oiSniperLevel.getAsDouble();
+    }
+    return sniperLevel;
+  }
 
+  public void setReversed(boolean reverse) {
+    joysticksReversed = reverse;
   }
 
   // Called once the command ends or is interrupted.
```

Comment:
interrupted is from the old framework and isn't needed. The code there can move to end.